### PR TITLE
Updates for mainly login stuff for seamless Widgetastic integration

### PR DESCRIPTION
**Pull request by mfalesni**

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit e2355dea2a63e1da682e94324d6c544c456bf1c2:

`utils/appliance/endpoints/ui.py`:
- :red_circle: Line 8:1: **F401** *'click' imported but unused*
- :red_circle: Line 8:1: **F401** *'elements' imported but unused*
- :red_circle: Line 8:1: **F401** *'execute_script' imported but unused*
- :red_circle: Line 8:1: **F401** *'handle_alert' imported but unused*
- :red_circle: Line 8:1: **F401** *'is_displayed' imported but unused*
- :red_circle: Line 8:1: **F401** *'text' imported but unused*
- :red_circle: Line 22:1: **F401** *'LocalProxy' imported but unused*

Please, rectify these issues :smirk: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 52e4bc1bd2b9784833cf50d707b3a4ebc1c7cccd:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

File: cfme/login.py [hunk 1]
```diff
@@ -60,13 +63,16 @@ def submit_login(self, method='click_on_login'):
             self.browser.execute_script('miqAjaxAuth();')
         else:
             raise ValueError('Unknown method {}'.format(method))
+        if self.flash.is_displayed:
+            self.flash.assert_no_error()
 
-    def log_in(self, username, password, method='click_on_login'):
+    def log_in(self, user, method='click_on_login'):
```
Comment (psav) [inline cfme/login.py]: Good! I wanted this to be a User object

File: cfme/tests/test_login.py [hunk 2]
```diff
@@ -1,50 +1,49 @@
...
-@pytest.mark.usefixtures("check_logged_out")
 def test_login(method):
     """ Tests that the appliance can be logged into and shows dashboard page. """
-    pytest.sel.get(pytest.sel.base_url())
-    assert not pytest.sel.is_displayed(dashboard.page.user_dropdown)
-    login.login_admin(submit_method=method)
-    assert pytest.sel.is_displayed(dashboard.page.user_dropdown), "Could not determine if logged in"
-    login.logout()
-    assert login.page.is_displayed()
+    appliance = get_or_create_current_appliance()
```
Comment (psav) [inline cfme/tests/test_login.py]: @RonnyPfannschmidt is this supposed to be 
```from utils.appliance import current_appliance```
and then use ```current_appliance```

File: utils/appliance/endpoints/ui.py [hunk 3]
```diff
@@ -310,19 +308,19 @@ def do_nav(self, _tries=0):
...
-            browser().quit()  # login.current_user() will be retained for next login
+            self.appliance.browser.quit_browser()
             logger.debug('browser killed on try {}'.format(_tries))
             # If given a "start" nav destination, it won't be valid after quitting the browser
             self.go(_tries)
 
     def go(self, _tries=0):
         _tries += 1
-        self.pre_navigate(_tries)
         self.appliance.browser.widgetastic.dismiss_any_alerts()
```
Comment (psav) [inline utils/appliance/endpoints/ui.py]: Can this dismissal go as the first thing inside the pre_navigate?

File: cfme/login.py (hunk 1 above)
Comment (mfalesni) [inline cfme/login.py]: Yes, I realized that too.

File: utils/appliance/endpoints/ui.py (hunk 3 above)
Comment (mfalesni) [inline utils/appliance/endpoints/ui.py]: I guess it can.

File: cfme/tests/test_login.py (hunk 2 above)
Comment (RonnyPfannschmidt) [inline cfme/tests/test_login.py]: i believe this test can use that object just fine

**Pull request merged**